x-pack/test/functional/services/observability/alerts.ts

```diff
@@ -109,21 +129,58 @@ export function ObservabilityAlertsProvider({ getPageObjects, getService }: FtrP
     return await testSubjects.findAllDescendant('alertsFlyoutDescriptionListDescription', flyout);
   };
 
+  const openRowActionsOverflowMenu = async (rowIndex: number) => {
+    const rows = await getTableCellsInRows();
+    const actionsOverflowButton = await testSubjects.findDescendant(
+      'alerts-table-row-action-more',
+      rows[rowIndex][ACTION_COLUMN_INDEX]
+    );
+    await actionsOverflowButton.click();
+  };
+
+  const setSingleAlertWorkflowStatus = async (
+    rowIndex: number,
+    workflowStatus: 'open' | 'acknowledged' | 'closed'
+  ) => {
+    await openRowActionsOverflowMenu(rowIndex);
+
+    if (workflowStatus === 'closed') {
+      await testSubjects.click('close-alert-status');
+    } else {
+      await testSubjects.click(`${workflowStatus}-alert-status`);
+    }
+
+    // wait for a confirmation toast (the css index is 1-based)
+    await toasts.getToastElement(1);
+    await toasts.dismissAllToasts();
+  };
+
+  const setWorkflowStatusFilter = async (workflowStatus: 'open' | 'acknowledged' | 'closed') => {
+    const buttonGroupButton = await testSubjects.find(
+      `workflow-status-filter-${workflowStatus}-button`
+    );
+    await buttonGroupButton.click();
+  };
+
   return {
     clearQueryBar,
-    typeInQueryBar,
-    submitQuery,
-    getTableCells,
-    getTableOrFail,
-    getNoDataStateOrFail,
-    openAlertsFlyout,
-    getAlertsFlyout,
-    getAlertsFlyoutTitle,
     closeAlertsFlyout,
-    navigateToTimeWithData,
+    getAlertsFlyout,
+    getAlertsFlyoutDescriptionListDescriptions,
+    getAlertsFlyoutDescriptionListTitles,
     getAlertsFlyoutOrFail,
+    getAlertsFlyoutTitle,
     getAlertsFlyoutViewInAppButtonOrFail,
-    getAlertsFlyoutDescriptionListTitles,
-    getAlertsFlyoutDescriptionListDescriptions,
+    getNoDataStateOrFail,
+    getTableCells,
+    getTableCellsInRows,
+    getTableColumnHeaders,
+    getTableOrFail,
+    navigateToTimeWithData,
+    openAlertsFlyout,
+    setSingleAlertWorkflowStatus,
+    setWorkflowStatusFilter,
+    submitQuery,
+    typeInQueryBar,
   };
```

Comment:
What would be the grouping criterion? Maybe `alerts_table` and `alerts_filters`? But then again, the filters filter the table. :shrug:

Reply:
To avoid conflicts with other PRs in progress, do we want to clean this up afterwards?

Reply:
Let's come back to this later! As to how we might group it, I think grouping by table actions and filter actions makes sense even if the two are related on the same page, they don't directly manipulate each other. 

In my understanding, in the test, you grab an instance of the page object for the AlertsPage, which has sub-components `table` and `filterBar`. You might write a test that takes an action on `filterBar` but does the assertion via `table`, yet the code for those two can live in different files since they don't interact with each other directly, but rather indirectly by being in the same page.